compiler/shared_ast/typing.ml

```diff
@@ -240,19 +244,19 @@ let polymorphic_op_type (op : ('a, Operator.polymorphic) A.operator Marked.pos)
   let it = lazy (UnionFind.make (TLit TInt, pos)) in
   let array a = lazy (UnionFind.make (TArray (Lazy.force a), pos)) in
   let ( @-> ) x y =
-    lazy (UnionFind.make (TArrow (Lazy.force x, Lazy.force y), pos))
```

Comment:
This is for polymorphic higher-order functions, like fold, reduce, map etc. Because its taking functions as arguments, we cannot use the same trick as for operators.ml.